homeassistant/components/tplink/light.py

```diff
@@ -312,19 +325,29 @@ def _handle_coordinator_update(self) -> None:
 class TPLinkSmartLightStrip(TPLinkSmartBulb):
     """Representation of a TPLink Smart Light Strip."""
 
-    device: SmartLightStrip
+    def __init__(
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        light_module: Light,
+        effect_module: LightEffect,
+    ) -> None:
+        """Initialize the switch."""
+        self._effect_module = effect_module
+        super().__init__(device, coordinator, light_module)
+
     _attr_supported_features = LightEntityFeature.TRANSITION | LightEntityFeature.EFFECT
 
     @callback
     def _async_update_attrs(self) -> None:
         """Update the entity's attributes."""
         super()._async_update_attrs()
-        device = self.device
-        if (effect := device.effect) and effect["enable"]:
-            self._attr_effect = effect["name"]
+        effect_module = self._effect_module
+        if effect_module.effect != LightEffect.LIGHT_EFFECTS_OFF:
+            self._attr_effect = effect_module.effect
```

Comment:
The common effect module returns a string for the current effect rather than a `dict`. If the effect is not enabled the value is `LightEffect.LIGHT_EFFECTS_OFF`.